modules/net-vpc/variables.tf

```diff
@@ -108,6 +108,50 @@ variable "peering_config" {
   default = null
 }
 
+variable "policy_based_routes" {
+  description = "Policy based routes, keyed by name."
+  type = map(object({
+    description         = optional(string, "Terraform-managed.")
+    labels              = optional(map(string))
+    priority            = optional(number)
+    next_hop_ilb_ip     = optional(string)
+    use_default_routing = optional(bool, false)
+    filter = optional(object({
+      ip_protocol = optional(string)
+      dest_range  = optional(string)
+      src_range   = optional(string)
+    }), {})
+    target = optional(object({
+      interconnect_attachment = optional(string)
+      tags                    = optional(list(string))
+    }), {})
+  }))
+  default  = {}
+  nullable = false
+  validation {
+    condition = alltrue([
+      for r in var.policy_based_routes :
+      contains(["TCP", "UDP", "ALL", null], r.filter.ip_protocol)
+      if r.filter.ip_protocol != null
+    ])
+    error_message = "Unsupported protocol for route."
+  }
+  validation {
+    condition = alltrue([
+      for r in var.policy_based_routes :
+      r.use_default_routing != (r.next_hop_ilb_ip != null)
```

Comment:
I can't parse this lol

Reply:
🤣 https://developer.hashicorp.com/terraform/language/expressions/operators#:~:text=Terraform%20does%20not%20have%20an%20operator%20for%20the%20%22exclusive%20OR

Reply:
Happy to! 
That would be
`(r.use_default_routing == true && r.next_hop_ilb_ip == null) || (r.use_default_routing && r.next_hop_ilb_ip != null)`

Would you prefer that?

Reply:
or

```hcl
(
  r.use_default_routing == true ? 1 : 0
  +
  r.next_hop_ib_ip != null ? 1 : 0
) == 1
```